### PR TITLE
parser: Support keyword `FIELDS` in `ALTER TABLE PARTITION BY` syntax

### PR DESCRIPTION
**Pull request by spongedu**
### What problem does this PR solve? 

To #464 , support keyword `FIELDS` in `ALTER TABLE PARTITION BY` syntax

### What is changed and how it works?

1. Use `FieldsOrColumns` instead of `COLUMNS` in `PartitionMethod`
2. Add related tests

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

N / A

Side effects

N / A

Related changes

N / A

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -2251,6 +2251,11 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"alter table t enable keys, comment = 'cmt' partition by hash(a)", true, "ALTER TABLE `t` ENABLE KEYS, COMMENT = 'cmt' PARTITION BY HASH (`a`) PARTITIONS 1"},
 		{"alter table t enable keys, comment = 'cmt', partition by hash(a)", false, ""},
 
+		{"alter table t partition by range FIELDS(a)", false, ""},
```
Comment (kennytm) [inline parser_test.go]: Please use a statement which can be parsed correctly here, e.g.

```suggestion
		{"alter table t partition by range FIELDS(a) (partition x values less than maxvalue)", true, "ALTER TABLE `t` PARTITION BY RANGE COLUMNS (`a`) (PARTITION `x` VALUES LESS THAN (MAXVALUE))"},
```

Comment (spongedu) [reply parser_test.go]: ok. I'' improve these tests

**Comment by leoppro** [comment]
LGTM

**Comment by leoppro** [comment]
please resolve the conflict

**Pull request merged**